modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/QuantumToolkit.java

```diff
@@ -1589,47 +1585,40 @@ public void run() {
                         // +-----------+-----------+  .  +-------+
                         // |     B     |     B     |  .  |   C   |
                         // +-----------+-----------+  .  +-------+
-
+                        int mTileWidth = computeTileSize(w, maxTextureSize);
+                        int mTileHeight = computeTileSize(h, maxTextureSize);
```

Comment:
I'll test it soon. Since these variables no longer change, maybe you could make them `final`?